### DISCUSSION TITLE
How do I modify a step and it's parent step Status?

### DISCUSSION THREAD

**Discussion by jgfmachado** [allure-framework/allure2#2162] [Questions & Support] [answered]
I see that the only option to change a step status is through `Allure.step(String, Status);`, but that doesn't change the status of the parent step.

For instance, this code below shows a passed "Step" as a green dropdown arrow, that includes a failed "Sub Step". But I'd like to have but "Step" and "Sub Step" marked as failed.
```
@Step("Step")
exampleMethod() {
   if(x.equals(false)) {
      Allure.step("Sub Step", Status.FAILED);
   }
}
```

Furthermore, shouldn't we be able to also add a status to "Lambda steps" like so,
```
exampleMethod() {
   if(x.equals(false)) {
      Allure.step("Step 1", (step) -> {
         Allure.step("Sub-step 1");
         Allure.step("Sub-step 2");
         },
      Status.FAILED);
   }
}
```

**Comment by baev** [comment, marked as answer]
There are a few options available: 

1. Throw an exception and process it at the parent step
2. Use Allure Lifecycle API to access the current running step:
    ```java
    Allure.getLifecycle().updateStep(step -> step.setStatus(Status.FAILED));
    ```
3. Post-process the report data using Lifecycle listener and re-calculate step statuses based on required conditions
